lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -15,12 +15,40 @@ class FirebaseAppDistributionAction < Action
       FIREBASECMD_ACTION = "appdistribution:distribute".freeze
       BASE_URL = "https://firebaseappdistribution.googleapis.com"
       PATH = "/v1alpha/apps/"
+      MAX_POLLING_RETRIES = 60
+      POLLING_INTERVAL_S = 2
+      CONNECTION = Faraday.new(url: BASE_URL) do |conn|
```

Comment:
I think your instinct to share this connection is good, and looks like it should work. However, using an all-caps variable name here makes this a "constant" variable, which is shared between all instances of the class. That seems like overkill, and I think it's better to keep one connection per instance of the class, to isolate the behavior. In order to do that you can use an "instance" variable, which starts with a "@" like `@connection` for example.

In this case it might be good to use a pattern I've seen folks use a lot:
```
def connection
  @connection ||= Faraday.new(...
end
```
And then when you want to use a connection you just use `connection`. This calls the method, which will either return the `@connection` instance variable if it exists, or will create it if it does not, and then return it. This has the added benefit of making it "lazy" as well, meaning it doesn't actually create the connection object until you use it. See more [here](http://blog.jayfields.com/2007/07/ruby-lazily-initialized-attributes.html).

Let me know if you want to chat this out.

Reply:
+1 Great description!